### PR TITLE
docs(samples): updated samples code to use async await

### PR DESCRIPTION
**Pull request by vijay-qlogic**
Fixes https://github.com/googleapis/google-cloud-node/issues/2869 (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: samples/batch.js [hunk 1]
```diff
@@ -40,21 +45,17 @@ function createQueryPartitions(instanceId, databaseId, identifier, projectId) {
...
-      const partitions = data[0];
-      console.log(
-        `Successfully created ${partitions.length} query partitions.`
-      );
-    })
-    .catch(err => {
-      console.error('ERROR:', err);
-    });
+  try {
+    const data = await transaction.createQueryPartitions(query);
```
Comment (JustinBeckwith) [inline samples/batch.js]: Here (and throughout) - it's a lot easier to just destructure the array during assignment:

```js
const [partitions] = await transaction.createQueryPartitions(query);
```

Then you can drop the next line all together :)

File: samples/batch.js [hunk 2]
```diff
@@ -40,21 +45,17 @@ function createQueryPartitions(instanceId, databaseId, identifier, projectId) {
...
+    const partitions = data[0];
+    console.log(`Successfully created ${partitions.length} query partitions.`);
+  } catch (err) {
```
Comment (JustinBeckwith) [inline samples/batch.js]: Let's just drop the `try/catch` stuff unless there's a finally we really need to execute.

File: samples/system-test/spanner.test.js [hunk 3]
```diff
@@ -172,493 +172,493 @@ test.serial(`should add a column to a table`, async t => {
...
-  async t => {
-    const results = await tools.runAsyncWithIO(
-      `${dmlCmd} deleteUsingPartitionedDml ${INSTANCE_ID} ${DATABASE_ID} ${PROJECT_ID}`,
-      cwd
-    );
-    const output = results.stdout + results.stderr;
-    t.regex(output, /Successfully deleted 5 records/);
-  }
-);
+// // update_data
```
Comment (JustinBeckwith) [inline samples/system-test/spanner.test.js]: Heh, assuming an update is coming here :)

**Comment by JustinBeckwith** [comment]
Looks like the system tests are still failin

**Comment by vijay-qlogic** [comment]
Yes, working to fix those failure

**Comment by JustinBeckwith** [comment]
@vijay-qlogic I sent you an invite to the googleapis org.  If you accept that, sample and system tests will run automatically on these PRs :)

**Comment by vijay-qlogic** [comment]
@JustinBeckwith  Yes, I accepted that invitation Thanks!, also enabled 2-factor authentication as well
Few tests are still failing for me, working on it to fix the issues, will update the PR as soon I resolve those

File: samples/transaction.js [hunk 4]
```diff
@@ -122,7 +119,7 @@ function readWriteTransaction(instanceId, databaseId, projectId) {
   const transferAmount = 200000;
   const minimumAmountToTransfer = 300000;
 
-  database.runTransaction((err, transaction) => {
+  database.runTransaction(async (err, transaction) => {
```
Comment (JustinBeckwith) [inline samples/transaction.js]: Having an async callback is very weird.  Should we just make the top level method this is under async instead?

**Pull request merged**